types/params.go

```diff
@@ -17,8 +18,9 @@ const (
 // ConsensusParams contains consensus critical parameters that determine the
 // validity of blocks.
 type ConsensusParams struct {
-	BlockSize      `json:"block_size_params"`
-	EvidenceParams `json:"evidence_params"`
+	BlockSize            `json:"block_size_params"`
```

Comment:
Thinking about this, we can probably drop the `Params` and `_params` suffixes from these fields.

Also I wonder if the ConsensusParams is a good place for `Version` to live? Then we don't need to add anything to EndBlock for the app to update the versions, its just part of the ConsensusParams.

As for this PR, maybe we should have a new struct to hold the pubkey types? Could be called `Signatures` or `Cryptography` or something ?

Reply:
This makes a lot of sense for app version imo.

Re new struct, sure I think Validators makes sense! I do think having the params suffix is nice for disambiguation when your looking at types.

Reply:
Cool. Let's go with `Validators`.

Yeh, `Params` makes sense for the types, but it's redundant for the field names. I guess here it's all embedded structs, but maybe it shouldn't be, and then we can have proper field names and struct type name distinctions.